app/addons/setup/actions.js

```diff
@@ -0,0 +1,369 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+import FauxtonAPI from "../../core/api";
+import 'whatwg-fetch';
+import {isInvalid} from "./helpers";
+import {
+  SETUP_ADD_NODE_TO_LIST,
+  SETUP_BIND_ADDRESS_ADDITIONAL_NODE,
+  SETUP_BIND_ADDRESS_FOR_SINGLE_NODE,
+  SETUP_NODE_COUNT,
+  SETUP_PORT_ADDITIONAL_NODE,
+  SETUP_PORT_FOR_SINGLE_NODE,
+  SETUP_REMOTE_ADDRESS_ADDITIONAL_NODE,
+  SETUP_RESET_ADDITIONAL_NODE,
+  SETUP_SET_CLUSTERSTATUS,
+  SETUP_SET_PASSWORD,
+  SETUP_SET_USERNAME
+} from "./actiontypes";
+import {get, rejectFetchError} from '../../core/ajax';
+import Api from "../auth/api";
+
+
+/**
+ * @typedef {Object} CredentialObject
```

Comment:
What are these comments for? Did you use to have an object here?

Reply:
It's a JsDoc type definition. IDE usually recognize those definitions and can provide information about object of that type.